app/src/main/java/ca/klapstein/baudit/activities/PatientHomeActivity.java

```diff
@@ -162,27 +160,73 @@ public void updateEmailDisplay(String email) {
         public ProblemViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
             CardView v = (CardView) LayoutInflater.from(viewGroup.getContext())
                     .inflate(R.layout.card_problem, viewGroup, false);
-            return new ProblemViewHolder(v); //Wrap it in a ViewHolder.
+            return new ProblemViewHolder(v); // Wrap it in a ViewHolder.
         }
 
         @Override
         public void onBindViewHolder(@NonNull final ProblemViewHolder viewHolder, int i) {
-            Problem problem = presenter.getProblemAt(i);
+            final Problem problem = presenter.getProblemAt(i);
             viewHolder.updateProblemTitleText(problem.getTitle());
             viewHolder.updateProblemDateText(problem.getTimeStamp());
             viewHolder.updateProblemDescriptionText(problem.getDescription());
+
             viewHolder.cardView.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    Intent intent = new Intent(
-                        PatientHomeActivity.this,
-                        ProblemActivity.class
-                    );
-                    intent.putExtra("problemId", viewHolder.getAdapterPosition()); // Test ID
-                    // TODO: Need a way to get the problem's ID to add to the intent
+                    Intent intent = new Intent(getApplicationContext(), ProblemActivity.class);
+                    intent.putExtra("problemPosition", viewHolder.getAdapterPosition());
+                    intent.putExtra("mode", "view");
```

Comment:
I think it would be best if these intent flags would be specified as public static final fields within their respective classes. Its safer, and cleaner.

Likely we will get code cleanliness bonus points.

Reply:
Agreed